public/app/features/transformers/extractFields/fieldExtractors.ts

```diff
@@ -3,7 +3,7 @@ import { Registry, RegistryItem } from '@grafana/data';
 import { FieldExtractorID } from './types';
 
 export interface FieldExtractor extends RegistryItem {
-  parse: (v: string) => Record<string, any> | undefined;
+  parse: (v: string, exp?: string, source?: string, transformationIndex?: number) => Record<string, any> | undefined;
```

Comment:
I really hate needing to add all these specific variables to the overall parse function. Is there a better way?

The only other transformation that has something like this is the json transformation's JSON Path, and the path logic happens after the parse, not inside it like what regex needs.

Reply:
I'm not familiar with the codebase, but I think unfortunately not. Unless we only allow named groups, we need a fallback, so index and source are required. I quite like how you named the fallback matches, btw.

FieldExtractor/extRegex: is not unique per transform, so you can't add fields there, right?